content/_data/upgrades/2-277-1.adoc

```diff
@@ -0,0 +1,50 @@
+Some older plugin releases will cause the Jenkins "Configure System" page to fail to save.
+Other older plugin releases will report Java runtime exceptions until they are upgraded to the most recent release.
+
+Popular plugins that need to be updated after upgrading to Jenkins 2.277.1 include:
+
+* plugin:branch-api[Branch API plugin]
+* plugin:envinject[Environment injector plugin]
+* plugin:favorite[Favorite plugin]
+* plugin:github-branch-source[GitHub branch source plugin]
+* plugin:ldap[LDAP plugin]
+* plugin:maven-plugin[Maven integration plugin]
+* plugin:parameterized-trigger[Parameterized trigger plugin]
+* plugin:token-macro[Token macro plugin]
+
+Use the Plugin Manager to update plugins after upgrading to Jenkins 2.277.1.
+
+==== Spring security update
+
+Jenkins 2.277.1 has replaced the outdated Acegi Security library with the Spring Security library.
+The most popular plugins have already been updated as necessary to support the change.
+Refer to the link:https://github.com/jenkinsci/jep/blob/master/jep/227/compatibility.adoc[Spring Security compatibility table] for questions related to specific plugins.
+
+Privately maintained plugins may need changes to adapt to the change from Acegi Security to Spring Security.
+Refer to Jesse Glick's link:/blog/2020/11/10/spring-xstream/[Spring Security blog post] for more details.
+
+==== XStream update
+
+Jenkins 2.277.1 has replaced its outdated copy of the XStream XML serialization library with a recent release of the official XStream library.
+The most popular plugins have already been updated as necessary to support the change.
+Refer to the link:https://github.com/jenkinsci/jep/blob/master/jep/228/compatibility.adoc[XStream compatibility table] for questions related to specific plugins.
+
+Privately maintained plugins may need changes to adapt to the change from the Jenkins XStream fork to the official XStream library.
+Refer to Jesse Glick's link:/blog/2020/11/10/spring-xstream/[XStream unfork blog post] for more details.
+
+// Intentionally not including this, since I'm not aware of any issues reported and don't know what the user would do if there is a problem.
+//
+// ==== JQuery update
+//
+// Jenkins 2.277.1 has replaced its outdated copy of the link:https://jquery.com/[JQuery user interface lirary] with the most recent release library.
+// The most popular plugins have already been updated as necessary to support the change.
+
+==== CVS plugin unbundled
+
+Jenkins no longer automatically installs the plugin:cvs[CVS plugin] on startup if a plugin depending on Jenkins (then Hudson) 1.340 or earlier is discovered.
+If you use a plugin that relies on the functionality provided by CVS plugin and manage plugins outside the Jenkins plugin manager, you will now need to ensure yourself that a recent release of CVS plugin is installed.
+Jenkins will attempt to load such plugins but may fail at any time during startup or afterwards with `ClassNotFoundException` or similar.
```

Comment:
Would just copy and slightly adapt https://www.jenkins.io/doc/upgrade-guide/2.204/#stop-bundling-maven-plugin-and-subversion-plugin-with-jenkins for the explanation. This is otherwise lacking context.

Reply:
Thanks for the pointer.  Done in ff47a1801f83a6035e8fd6b5c957aa878771f904